core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala

```diff
@@ -73,7 +73,7 @@ private[spark] class MetricsSystem private (
   private[this] val metricsConfig = new MetricsConfig(conf)
 
   private val sinks = new mutable.ArrayBuffer[Sink]
-  private val sources = new mutable.ArrayBuffer[Source]
+  val sources = new mutable.ArrayBuffer[Source]
```

Comment:
@nssalian This has not been addressed. This should not be public and see the link to see how to keep it private and yet access it in tests.